devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -291,58 +302,73 @@ class HumanStateProvider::impl::CmdParser : public yarp::os::PortReader
     std::string parentLinkName;
     std::string childLinkName;
+    // variables for manual calibration
+    std::atomic<double> roll;
+    std::atomic<double> pitch;
+    std::atomic<double> yaw;
 
     void resetInternalVariables()
     {
         parentLinkName = "";
         childLinkName = "";
-        cmdStatus = false;
-        cmdErease = false;
+        cmdStatus = rpcCommand::empty;
     }
 
     bool read(yarp::os::ConnectionReader& connection) override
     {
         yarp::os::Bottle command, response;
         if (command.read(connection)) {
-
             if (command.get(0).asString() == "help") {
                 response.addVocab(yarp::os::Vocab::encode("many"));
                 response.addString("The following commands can be used to apply a secondary calibration assuming the subject is in the zero configuration of the model for the calibrated links. \n");
-                response.addString("Enter <calibrate> to apply a secondary calibration for all the links \n");
+                response.addString("Enter <calibrateAll> to apply a secondary calibration for all the links \n");
                 response.addString("Enter <calibrate <linkName>> to apply a secondary calibration for the given link \n");
-                response.addString("Enter <calibrate <parentLinkName> <childLinkName>> to apply a secondary calibration for the given chain \n");
+                response.addString("Enter <setRotationOffset <linkName> <r p y [deg]>> to apply a secondary calibration for the given link using the given rotation offset (defined using rpy)\n");
+                response.addString("Enter <calibrateSubTree <parentLinkName> <childLinkName>> to apply a secondary calibration for the given chain \n");
+                response.addString("Enter <calibrateRelativeLink <parentLinkName> <childLinkName>> to apply a secondary calibration for the child link using the parent link as reference \n");
                 response.addString("Enter <reset <linkName>> to remove secondary calibration for the given link \n");
                 response.addString("Enter <reset> to remove all the secondary calibrations");
             }
-            else if (command.get(0).asString() == "calibrate" && !command.get(1).isNull() && !command.get(2).isNull()) {
+            else if (command.get(0).asString() == "calibrateRelativeLink" && !command.get(1).isNull() && !command.get(2).isNull()) {
                 this->parentLinkName = command.get(1).asString();
                 this->childLinkName = command.get(2).asString();
-                response.addString("Entered command <calibrate> is correct, setting the offset for the chain from " + this->parentLinkName + " to " + this->childLinkName);
-                this->cmdStatus = true;
+                response.addString("Entered command <calibrateRelativeLink> is correct, trying to set offset to " + this->childLinkName + " using " + this->parentLinkName + " as reference");
```

Comment:
@lrapetti I guess the message is "Entered command <calibrateRelativeLink> is correct, trying to set offset ~~to~~ **of** " + this->childLinkName + " using " + this->parentLinkName + " as reference

Reply:
Done https://github.com/robotology/human-dynamics-estimation/pull/211/commits/e7bbccb8cef6e44580587fa63930c23e05b4bb51